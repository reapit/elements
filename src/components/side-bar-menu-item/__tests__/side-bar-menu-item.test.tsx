import { render } from '@testing-library/react'
import { SideBarMenuItem } from '..'

describe('SideBarMenuItem', () => {
  it('should render item and match snapshot', () => {
    const { asFragment } = render(
      <SideBarMenuItem icon={<div />} href="#">
        a link
      </SideBarMenuItem>,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render active item and match snapshot', () => {
    const { asFragment } = render(
      <SideBarMenuItem icon={<div />} isActive href="/#">
        a link
      </SideBarMenuItem>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
