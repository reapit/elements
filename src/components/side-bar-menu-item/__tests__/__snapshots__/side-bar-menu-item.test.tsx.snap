// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SideBarMenuItem > should render anchor item and match snapshot 1`] = `
<DocumentFragment>
  <li>
    <a
      class="mocked-styled-3 el-side-bar-menu-item-anchor"
      href="#"
    >
      <span
        class="mocked-styled-1 el-side-bar-menu-item-icon"
      >
        <div />
      </span>
      <span
        class="mocked-styled-0 el-side-bar-menu-item-text"
      >
        a link
      </span>
    </a>
  </li>
</DocumentFragment>
`;

exports[`SideBarMenuItem > should render button item with isActive props and match snapshot 1`] = `
<DocumentFragment>
  <li>
    <button
      aria-current="page"
      class="mocked-styled-2 el-side-bar-menu-item-button"
    >
      <span
        class="mocked-styled-1 el-side-bar-menu-item-icon"
      >
        <div />
      </span>
      <span
        class="mocked-styled-0 el-side-bar-menu-item-text"
      >
        a button
      </span>
    </button>
  </li>
</DocumentFragment>
`;
